src/screens/runPlanSlide/RunPlanSlideScreen.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+import { NavigationInjectedProps } from 'react-navigation';
+import { RNFirebase } from 'react-native-firebase';
+
+import { StyledText, Card, FlatButton } from 'components';
+import { palette, typography } from 'styles';
+import { Student, PlanItem } from 'models'
+import { i18n } from 'locale';
+import { PlanSlideItem } from './PlanSlideItem';
+
+interface State {
+  planItems: PlanItem[];
+  pageNumber: number;
+  student: Student;
+}
+
+export class RunPlanSlideScreen extends React.PureComponent<
+  NavigationInjectedProps> {
+    planItemsRef: any;
+    unsubscribePlanItems: any;
+    studentRef: any;
+    unsubscribeStudent: any;
+    static navigationOptions = {
+      header: null,
+    };
+    state = {
+      planItems: [],
+      pageNumber: 0,
+      student: this.props.navigation.getParam('student'),
+    };
+
+    componentDidMount() {
+      const plan = this.props.navigation.getParam('plan');
+      this.planItemsRef = plan.getPlanItemsRef();
+      this.unsubscribePlanItems = this.planItemsRef.onSnapshot(this.handlePlanItemsChange);
+      this.studentRef = this.state.student.getStudentRef();
+      this.unsubscribeStudent = this.studentRef.onSnapshot(this.handleStudentChange);
+    }
+  
+    handlePlanItemsChange = (
+      querySnapshot: RNFirebase.firestore.QuerySnapshot,
+    ) => {
+      const planItems: PlanItem[] = querySnapshot.docs.map(doc =>
+        Object.assign(new PlanItem(), {
+          id: doc.id,
+          ...doc.data(),
+        }),
+      );
+      this.setState({ planItems });
+    };
+  
+    handleStudentChange = (documentSnapshot: RNFirebase.firestore.DocumentSnapshot) => {
+      if(documentSnapshot.exists){
+        const student = Object.assign(new Student(), {
+          id: documentSnapshot.id,
+          ...documentSnapshot.data(),
+        });
+        this.setState({ student });
+      }
+    }
+  
+    componentWillUnmount() {
+      this.unsubscribePlanItems();
+      this.unsubscribeStudent();
+    }
+  
+    nextPage = () => {
+      if(this.state.pageNumber + 1 < this.state.planItems.length)
+        this.setState({ pageNumber: this.state.pageNumber + 1 });
+    }
+
+    render() {
+      return (
+        this.state.planItems.length
+        ? <View style={styles.container}>
```

Comment:
What do you think about wrapping this in two methods, e.g. "renderList", and "renderLoader" ?